scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -622,6 +647,28 @@ class FormatWriter(formatOps: FormatOps) {
     }
   }
 
+  private def prepareAlignmentInfo(
+      block: Vector[Array[FormatLocation]],
+      separatorLengthGaps: Array[Int],
+      column: Int
+  ): Vector[AlignmentUnit] =
+    block.zipWithIndex.map {
```

Comment:
why not `block.zip(separatorLengthGaps` and remove the `lineIndex` field below?

Reply:
Similar motivation https://github.com/scalameta/scalafmt/pull/1759#discussion_r386346282